### PR TITLE
python dataset processor: initial settings and read csv

### PR DESCRIPTION
**Pull request by rugeli**
Problem
=======
What is the problem this work solves, including
an initial proposal for #91

Solution
========
What I/we did to solve this problem
- created a python directory and a command line `create-dataset` for data processing
- functionalities that load and convert the input data, and create the dataset folder and files
- skeletons for individual handlers to write to the files - `feature_defs.json`, `dataset.json`, `cell_feature_analysis.json` and `image-settings.json`

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. run command line: `npm run create-dataset -- dataset-processor-python/input_examples/test_data.csv`

### CODE REVIEW

**Comment by rugeli** [comment]
This is a draft for the initial design of the python dataset processing, drawing from the ideas discussed by Megan and Dan. Please feel free to share comments, suggest modifications, or bring any ideas you might have. We appreciate your collaborations on this!

File: dataset-processor-python/data_loader.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        super().__init__(path, dataset_name)
+
+    def get_inputs(self):
+        """
+        get the inputs from the user
+        """
+        # version is required, title and description are optional
+        # Q: should we have a default title and description?
+        # return version, title, description
+        pass
```
Comment (ShrimpCryptid) [inline dataset-processor-python/data_loader.py:43]: Is this getting command line inputs (prompting for user text) or from parameters?

**Review by ShrimpCryptid — APPROVED** [review]
Looks good! I ran the test command and it output a JSON file as expected.

File: dataset-processor-python/data_loader.py (hunk 1 above)
Comment (rugeli) [reply dataset-processor-python/data_loader.py:43]: As of now, we are gathering user inputs through the command line prompts. We might revise this approach depending on how much data is required from the user.

File: dataset-processor-python/data_loader.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
+import os
+import sys
+import csv
+import json
+
+
+class DataLoader:
+    def __init__(self, path, dataset_name):
+        self.path = path
+        self.dataset_name = dataset_name
```
Comment (meganrm) [inline dataset-processor-python/data_loader.py:10]: add `self.initial_data = None` here

File: dataset-processor-python/data_loader.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        converts the csv file to json format
+        """
+        try:
+            with open(self.path, "r") as f:
+                reader = csv.DictReader(f)
+                data = list(reader)
+            json_filename = self.dataset_name + ".json"
+            full_path = os.path.join(os.path.dirname(self.path), json_filename)
+            with open(full_path, "w") as f:
+                json.dump(data, f, indent=4)
```
Comment (meganrm) [inline dataset-processor-python/data_loader.py:23]:
```suggestion
            self.initial_data = data
```

File: dataset-processor-python/data_loader.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import csv
+import json
+
+
+class DataLoader:
+    def __init__(self, path, dataset_name):
+        self.path = path
+        self.dataset_name = dataset_name
+
+    def csv_to_json(self):
```
Comment (meganrm) [inline dataset-processor-python/data_loader.py:12]: change this to `read_csv`

**Review by meganrm — APPROVED** [review]
after you do the few suggestions I commented

**Pull request merged**